saleor/tests/e2e/orders/test_order_calculate_simple_tax_based_on_shipping_address_tax_class.py

```diff
@@ -0,0 +1,245 @@
+import pytest
+
+from ..channel.utils import create_channel
+from ..product.utils.preparing_product import prepare_product
+from ..shipping_zone.utils import (
+    create_shipping_method,
+    create_shipping_method_channel_listing,
+    create_shipping_zone,
+)
+from ..taxes.utils import (
+    get_tax_configurations,
+    update_country_tax_rates,
+    update_tax_configuration,
+)
+from ..utils import assign_permissions
+from ..warehouse.utils import create_warehouse
+from .utils import (
+    draft_order_complete,
+    draft_order_create,
+    draft_order_update,
+    order_lines_create,
+)
+
+
+def prepare_shop_with_few_shipping_zone_countries(
+    e2e_staff_api_client,
+    shipping_price,
+):
+    warehouse_data = create_warehouse(e2e_staff_api_client)
+    warehouse_id = warehouse_data["id"]
+    channel_slug = "test"
+    warehouse_ids = [warehouse_id]
+    channel_data = create_channel(
+        e2e_staff_api_client,
+        slug=channel_slug,
+        warehouse_ids=warehouse_ids,
+    )
+    channel_id = channel_data["id"]
+    channel_ids = [channel_id]
+    shipping_zone_data = create_shipping_zone(
+        e2e_staff_api_client,
+        countries=["CZ", "DE", "US"],
+        warehouse_ids=warehouse_ids,
+        channel_ids=channel_ids,
+    )
+    shipping_zone_id = shipping_zone_data["id"]
+
+    shipping_method_data = create_shipping_method(
+        e2e_staff_api_client, shipping_zone_id
+    )
+    shipping_method_id = shipping_method_data["id"]
+    create_shipping_method_channel_listing(
+        e2e_staff_api_client,
+        shipping_method_id,
+        channel_id,
+        price=shipping_price,
+    )
+
+    return (
+        warehouse_id,
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+        shipping_price,
+    )
+
+
+def prepare_tax_configuration(
+    e2e_staff_api_client,
+    channel_slug,
+    shipping_country_code,
+    shipping_country_tax_rate,
+    billing_country_code,
+    billing_country_tax_rate,
+):
+    tax_config_data = get_tax_configurations(e2e_staff_api_client)
+    channel_tax_config = tax_config_data[0]["node"]
+    assert channel_tax_config["channel"]["slug"] == channel_slug
+    tax_config_id = channel_tax_config["id"]
+
+    tax_config_data = update_tax_configuration(
+        e2e_staff_api_client,
+        tax_config_id,
+        charge_taxes=True,
+        tax_calculation_strategy="FLAT_RATES",
+        display_gross_prices=True,
+        prices_entered_with_tax=True,
```

Comment:
This flag should be false. Intentionally, in test cases, we create an order once with the setting prices_entered_with_tax=True, and in another test prices_entered_with_tax=False